ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -474,16 +474,37 @@ private StreamToTableJoiner(
 
       switch (joinNode.joinType) {
         case LEFT:
-          return leftTable.leftJoin(
-              rightTable,
-              joinNode.getKeyColumnName(),
-              contextStacker);
+          if (true) { // check if PK or FK
+            return leftTable.leftJoin(
+                rightTable,
+                joinNode.getKeyColumnName(),
+                contextStacker
+            );
+          } else {
+            return leftTable.foreignKeyJoin(
+                rightTable,
+                null, // fk-join column
+                contextStacker
+            );
+          }
         case INNER:
-          return leftTable.join(
-              rightTable,
-              joinNode.getKeyColumnName(),
-              contextStacker);
+          if (true) { // check if PK or FK
```

Comment:
So you would call `joinKey.isForeignKey()`, and then get the fk column with `((ForeignJoinKey) joinKey).getForeignKeyColumn()`

Reply:
Correct (assuming we don't change #7522 that adds those methods)